ADD  futur_usage stripe.class.php

ptibogxiv opened the pull request:
more compatibility with 3Dsecure with saved payment intent (need les authentication)

WIP: add or not a setting in admin  or payment for the choice

#11523

ptibogxiv committed:
Update stripe.class.php

ptibogxiv committed:
Update stripe.class.php

ptibogxiv committed:
Update stripe.class.php

ptibogxiv commented:
for on session, we need to add information to customer 
[https://stripe.com/docs/payments/cards/reusing-cards#mandates]

need more compliance to GDPR too with payment form with payment intent

eldy commented on htdocs/stripe/class/stripe.class.php:
```diff
@@ -392,7 +392,7 @@ public function getPaymentIntent($amount, $currency_code, $tag, $description = '
     		    "payment_method_types" => array("card"),
     		    "description" => $description,
     		    "statement_descriptor" => dol_trunc($tag, 10, 'right', 'UTF-8', 1),     // 22 chars that appears on bank receipt (company + description)
-    			//"save_payment_method" => true,
```
Why removing save_payment_method = true ?
When it was not set, the credit card was not saved on stripe side.

ptibogxiv commented:
Payment intent will impact all payment from 14 september. stripe change method for saving card for less authentication for off session payment (very new code). API have changed :/

[https://stripe.com/docs/payments/cards/saving-cards]

i will some other PR for more compatibility and use this card from Dolibarr UI or futur online shop or stripe terminal with POS

eldy committed:
Keep save_payment_method to true.

Keep it to true because i see no reason to not save the credit card on Stripe side.

trag-stripe commented:
Thanks for supporting SCA @eldy for customers that need paymentintents! When the work is completed we would love to feature Dolibarr on https://stripe.com/works-with/sca-ready

eldy commented:
@trag-stripe We are ready with v10 final release. Just one parameter to set to switch. We provide 2 different Stripe implementations, both ready for SCA. I have just updated the wiki page for Stripe features in Dolibarr ERP CRM. See bottom of page https://wiki.dolibarr.org/index.php/Module_Stripe

There is no need of extension. It is available by default into Dolibarr ERP CRM v10.
How can we be added to the list of SCA ready ?
Our application can is related to following categories:
CRM, Accounting, Business Management, Customer support, Point of sale, Shipping, Inventory management, Invoicing

trag-stripe commented:
great to hear, thank you!
Can you message me at {{ trag at stripe dot com }} and I'll help you get into the SCA ready list?

Talk soon
